api/tests/Feature/PoolApplicationTest.php

```diff
@@ -15,100 +16,183 @@
+    protected $createMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation createApplication($userId: ID!, $poolId: ID!){
+            createApplication(userId: $userId, poolId: $poolId) {
+                user {
+                    id
+                }
+                pool {
+                    id
+                }
+                status
+            }
+        }
+    ';
+
+    protected $archiveMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation archiveApplication($id: ID!) {
+            archiveApplication(id: $id) {
+                archivedAt
+            }
+        }
+    ';
+
+    protected $deleteMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation deleteApplication($id: ID!) {
+            deleteApplication(id: $id)
+        }
+    ';
+
+    protected $submitMutationDocument =
+    /** @lang GraphQL */
+    '
+        mutation submitTest($id: ID!, $sig: String!) {
+            submitApplication(id: $id, signature: $sig) {
+                submittedAt
+                signature
+                status
+            }
+        }
+    ';
 
     protected function setUp(): void
     {
         parent::setUp();
+
+        // Run necessary seeders
+        $this->seed(ClassificationSeeder::class);
+        $this->seed(GenericJobTitleSeeder::class);
+        $this->seed(SkillFamilySeeder::class);
+        $this->seed(SkillSeeder::class);
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->setUpFaker();
         $this->bootRefreshesSchemaCache();
+
+        Team::factory()->create([
+            'name' => $this->teamName,
+        ]);
+
+        $this->guestUser = User::factory()->create([
+            'email' => 'guest-user@test.com',
+            'sub' => 'guest-user@test.com',
+        ]);
+        $this->guestUser->syncRoles([
+            "guest",
+        ]);
+
+        $this->applicantUser = User::factory()->create([
+            'email' => 'applicant-user@test.com',
+            'sub' => 'applicant-user@test.com',
+        ]);
+        $this->applicantUser->syncRoles([
+            "guest",
+            "base_user",
+            "applicant"
+        ]);
+        // Add generic job title for submission
+        $this->applicantUser->expectedGenericJobTitles()->sync([GenericJobTitle::first()->id]);
+
+        $this->responderRole = User::factory()->create([
+            'email' => 'admin-user@test.com',
+            'sub' => 'admin-user@test.com',
+        ]);
+        $this->responderRole->syncRoles([
+            "guest",
+            "base_user",
+            "applicant",
+            "request_responder"
+        ]);
+
+        $this->teamUser = User::factory()->create([
+            'email' => 'team-user@test.com',
+            'sub' => 'team-user@test.com',
+        ]);
+        $this->teamUser->syncRoles([
+            "guest",
+            "base_user",
+            "applicant"
+        ]);
+        $this->teamUser->attachRole("pool_operator", $this->teamName);
```

Comment:
Have you created this team, before attaching it?

Reply:
Yup, however I realized I was doing it a little higher up which could cause some confusion. 

Cleaned up in [reorganize pool application setup](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5815/commits/a89751028a35c29da7172912a51399b58e829e78)